tests/test_install.py

```diff
@@ -126,6 +128,21 @@ def test_windows_entry_point():
     cwd = getcwd()
     chdir(tmp_dir)
     original_prefix = "C:\\BogusPrefix\\python.exe"
+    if not os.environ.get("PYTHONPATH"):
+        # NOTE: This is usually set by the CI or dev script config. This is a fallback.
+        # We need to set this so Python loads the dev version of 'conda', usually taken
+        # from the conda/ in the working directory (e.g. the root of the cloned repo in CI).
+        # Otherwise, it will import the one installed in the base environment, which might
+        # have not been overwritten with `pip install -e . --no-deps`. This doesn't happen
+        # in other tests because they run with the equivalent of `python -m conda`. However,
+        # this family of tests relies on conda (shell function) which calls conda (Python entry
+        # point). When a script is called this way, it bypasses the automatic "working directory
+        # is first on sys.path" behavior you find in `python -m` style calls. See
+        # https://docs.python.org/3/library/sys_path_init.html for details.
+        env = os.environ.copy()
+        env["PYTHONPATH"] = CONDA_SOURCE_ROOT
+    else:
+        env = os.environ
```

Comment:
isn't this a use case for `monkeypatch.setenv`?

Reply:
We need it in the subprocess only if it's not set. I think the subprocess will inherit the parent env in Unix, but not in Windows. At least I remember a conversation with @marcoesters in constructor or menuinst where he was chasing a strange bug that ended up being due to `env` not being passed as expected.

Reply:
hm can we track that down? per Python docs there is no difference between Unix and Windows re. env inheritance in subprocesses?

inline with the [autouse fixture suggestion](https://github.com/conda/conda/pull/14475#discussion_r1907591471) I'm thinking we only need to do the following:

```python
@pytest.fixture(scope="session", autouse=True)
def PYTHONPATH():
    if "PYTHONPATH" in os.environ:
        return

    with pytest.MonkeyPatch.context() as monkeypatch:
        monkeypatch.setenv("PYTHONPATH", CONDA_SOURCE_ROOT)
        yield
```

Reply:
@jaimergp, I remember this vaguely. From what I can track down, the subprocess module handles copying explicitly: https://github.com/python/cpython/blob/3.12/Lib/subprocess.py#L1757-L1758

For Windows, it defers to `_winapi` where I can only find an explicit copy for non-`None` values: https://github.com/python/cpython/blob/3.12/Modules/_winapi.c#L1272-L1277

Otherwise, `wenvironment` is `NULL`. However, I do not know what is done under the hood when the process opens. I did notice a difference when setting `env` to `None` and explicitly copying the entire environment. 

However, I don't think I used `monkeypatch` back then. This is the PR in question: https://github.com/conda/constructor/pull/771

I'd say if monkeypatch works, use monkeypatch.

Reply:
That autouse fixture is neat. Let me give it a try.